include/envoy/http/header_map.h

```diff
@@ -525,19 +526,27 @@ class HeaderMap {
    */
   virtual bool empty() const PURE;
 
+  void logState(std::ostream& os, int indent_level = 0) const {
+    using IterateData = std::pair<std::ostream*, const char*>;
+    const char* spaces = spacesForLevel(indent_level);
+    IterateData iterate_data = std::make_pair(&os, spaces);
+    iterate(
+        [](const HeaderEntry& header, void* context) -> HeaderMap::Iterate {
+          auto* data = static_cast<IterateData*>(context);
+          *data->first << data->second << "'" << header.key().getStringView() << "', '"
+                       << header.value().getStringView() << "'\n";
+          return HeaderMap::Iterate::Continue;
+        },
+        &iterate_data);
+  }
```

Comment:
Not a huge fan of putting this in the header, can we move this to an implementation? Also, I think it's a bit weird talking about `dumpState` on a relatively pure data structure like `HeaderMap`, and doing indent futzing in a pure header file is a bit of a code smell.